fastai/datasets.py

```diff
@@ -180,20 +186,22 @@ def url2name(url): return url.split('/')[-1]
 def url2path(url, data=True, ext:str='.tgz'):
     "Change `url` to a path."
     name = url2name(url)
-    return datapath4file(name, ext=ext) if data else modelpath4file(name, ext=ext)
-def _url2tgz(url, data=True, ext:str='.tgz'):
-    return datapath4file(f'{url2name(url)}{ext}', ext=ext) if data else modelpath4file(f'{url2name(url)}{ext}', ext=ext)
+    return datapath4file(name, ext=ext, archive=False) if data else modelpath4file(name, ext=ext)
+def _url2tgz(url, data=True, ext: str = '.tgz'):
+    "Gets archival path from `url`"
+    return datapath4file(f'{url2name(url)}{ext}', ext=ext, archive=True) if data else modelpath4file(f'{url2name(url)}{ext}', ext=ext)
 
 def modelpath4file(filename, ext:str='.tgz'):
     "Return model path to `filename`, checking locally first then in the config file."
     local_path = URLs.LOCAL_PATH/'models'/filename
     if local_path.exists() or local_path.with_suffix(ext).exists(): return local_path
     else: return Config.model_path()/filename
 
-def datapath4file(filename, ext:str='.tgz'):
+def datapath4file(filename, archive:bool, ext:str='.tgz'):
```

Comment:
This is the line causing problems in the tests (so potentially for users that use this function directly). Please add a default to that argument consistent with the current behavior (`archive=False`).